assemblies/resources/bin/check_ports

```diff
@@ -0,0 +1,74 @@
+#!/bin/bash
+# Licensed to The Apereo Foundation under one or more contributor license
+# agreements. See the NOTICE file distributed with this work for additional
+# information regarding copyright ownership.
+#
+#
+# The Apereo Foundation licenses this file to you under the Educational
+# Community License, Version 2.0 (the "License"); you may not use this file
+# except in compliance with the License. You may obtain a copy of the License
+# at:
+#
+#   http://opensource.org/licenses/ecl2.txt
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+# -----------------------------------------------------------------------
+# check if http or https port is available to run opencast otherwise exit
+
+PROGNAME=$(basename "$0")
+
+if [ "$PROGNAME" = "start-opencast" ]; then
+
+  if [ -d "$1" ]; then
+    SETENV_ETC="$1"
+  else
+    if [ -d "${KARAF_BASE}" ]; then
+      SETENV_ETC="${KARAF_BASE}/etc"
+    else
+      SETENV_ETC="etc"
+    fi
+  fi
+
+  check_port() {
+    if [ "$1" ] ; then
+      hexPort=$(printf ":%04X " "$1")
+      for i in /proc/net/*; do
```

Comment:
This will not work on not linux machines, for example OSX (there is no /proc filesystem)

Reply:
is there any way to find out from the system which ports are in use on a OSX?

Reply:
can I rely on netstat or lsof being available on OSX?

Reply:
Apple ships both (`/usr/sbin/lsof`, `/usr/sbin/netstat`) although I'm not sure if you have to install the dev tools. I will check with a fresh macOS install.

Reply:
So a fresh install of macOS includes the tools in the mentioned locations.

Reply:
Apart from the problem below, the function works on macOS.